neuralop/layers/segment_csr.py

```diff
@@ -0,0 +1,57 @@
+"""
+Replacement for torch_scatter.segment_csr
+
+"""
+from typing import Literal
+import torch
+
+def segment_csr(src: torch.Tensor, indptr: torch.Tensor, reduce: Literal['mean', 'sum']):
+    """segment_csr reduces all entries of a CSR-formatted 
+    matrix by summing or averaging over neighbors. 
+
+    Used to reduce features over neighborhoods 
+    in neuralop.layers.IntegralTransform
+    
+    Parameters
+    ----------
+    src : torch.Tensor
+        tensor of features for each point
+    indptr : torch.Tensor
+        splits representing start and end indices 
+        of each neighborhood in src
+    reduce : Literal['mean', 'sum']
+        how to reduce a neighborhood. if mean,
+        reduce by taking the average of all neighbors.
+        Otherwise take the sum. 
+    """
+    if reduce not in ['mean', 'sum']:
+        raise ValueError("reduce must be one of \'mean\', \'sum\'")
+    
+    if torch.backends.cuda.is_built():
+        """only import torch_scatter when cuda is available"""
+        import torch_scatter.segment_csr as scatter_segment_csr
+        return scatter_segment_csr(src, indptr, reduce)
```

Comment:
Hmm cuda being built doesn't necessarily mean torch_scatter will be available? Shouldn't we check for both?

Reply:
Logic is changed.

Reply:
@dhpitt could you please check that your segment_csr works with backwards. You're doing some inplace operations so I'm not sure.